src/webhooks/registry.ts

```diff
@@ -261,20 +261,31 @@ const WebhooksRegistry: RegistryInterface = {
       deliveryMethod === DeliveryMethod.Http
         ? `${Context.HOST_SCHEME}://${Context.HOST_NAME}${path}`
         : path;
-    const checkResult = (await client.query({
-      data: buildCheckQuery(topic),
-    })) as {body: WebhookCheckResponse};
-    let webhookId: string | undefined;
-    let mustRegister = true;
-    if ('errors' in checkResult.body) {
-      registerReturn[topic] = {
-        success: false,
-        result: checkResult.body,
-      };
+
+    let checkResult: {body: WebhookCheckResponse};
+    try {
+      checkResult = (await client.query({
+        data: buildCheckQuery(topic),
+      })) as {body: WebhookCheckResponse};
+    } catch (error) {
+      if (error instanceof ShopifyErrors.GraphqlQueryError) {
+        registerReturn[topic] = {
+          success: false,
+          result: error.response,
+        };
+      } else {
+        registerReturn[topic] = {
+          success: false,
+          result: {},
+        };
+      }
```

Comment:
Nit: we could simplify this a bit so that we only need to write the response object once:
```suggestion
      const result = (error instanceof ShopifyErrors.GraphqlQueryError) ? error.response : {};
      registerReturn[topic] = { success: false, result };
```